resources/templates/lunr/build-index.js

```diff
@@ -0,0 +1,126 @@
+var path = require("path");
+var fs = require("fs");
+var lunr = require("lunr");
+var cheerio = require("cheerio");
+
+
+// Change these constants to suit your needs
+const HTML_FOLDER = "public/posts-output";  // folder with your HTML files
+// Valid search fields: "title", "description", "keywords", "body"
+const SEARCH_FIELDS = ["title", "description", "keywords", "body"];
+const EXCLUDE_FILES = ["search.html"];
+const MAX_PREVIEW_CHARS = 275;  // Number of characters to show for a given search result
+const OUTPUT_INDEX = "themes/nucleus/js/index/lunr_index.js";  // Index file
+
+
+function isHtml(filename) {
+    lower = filename.toLowerCase();
+    return (lower.endsWith(".htm") || lower.endsWith(".html"));
+}
+
+
+function findHtml(folder) {
+    if (!fs.existsSync(folder)) {
+        console.log("Could not find folder: ", folder);
+        return;
+    }
+
+    var files = fs.readdirSync(folder);
+    var htmls = [];
+    for (var i = 0; i < files.length; i++) {
+        var filename = path.join(folder, files[i]);
+        var stat = fs.lstatSync(filename);
+        if (stat.isDirectory()) {
+            var recursed = findHtml(filename);
+            for (var j = 0; j < recursed.length; j++) {
+                recursed[j] = path.join(files[i], recursed[j]).replace(/\\/g, "/");
+            }
+            htmls.push.apply(htmls, recursed);
+        }
+        else if (isHtml(filename) && !EXCLUDE_FILES.includes(files[i])) {
+            htmls.push(files[i]);
+        };
+    };
+    return htmls;
+};
+
+
+function readHtml(root, file, fileId) {
+    var filename = path.join(root, file);
+    var txt = fs.readFileSync(filename).toString();
+    var $ = cheerio.load(txt);
+    /* remove blog title from title field */
+    var title = $("title").text();
+
+    if (typeof title == 'undefined') title = file;
+    var description = $("meta[name=description]").attr("content");
+    if (typeof description == 'undefined') description = "";
+    var keywords = $("meta[name=keywords]").attr("content");
+    if (typeof keywords == 'undefined') keywords = "";
+    var body = $("body").text()
+    if (typeof body == 'undefined') body = "";
+    var data = {
+        "id": fileId,
+        "link": "posts-output/"+file,
+        "t": title,
+        "d": description,
+        "k": keywords,
+        "b": body
+    }
+    return data;
+}
+
+
+function buildIndex(docs) {
+    var idx = lunr(function () {
+        this.ref('id');
+        for (var i = 0; i < SEARCH_FIELDS.length; i++) {
+            this.field(SEARCH_FIELDS[i].slice(0, 1));
```

Comment:
Perhaps it would be nice to add a boost to the title field and perhaps also a small one to the description. For simplicity:

```js
var SEARCH_FIELD_BOOSTS = {title: 2, description: 1.3}; // far above at the global scope
...
var boost = SEARCH_FIELD_BOOSTS[SEARCH_FIELDS[i]] || 1;
this.field(SEARCH_FIELDS[i].slice(0, 1), { boost });
```

Reply:
good stuff